api/stacks.ts

```diff
@@ -141,6 +155,8 @@ export async function fetchStxAddressData(
     getConfirmedTransactions({
       stxAddress,
       network,
+      offset: offset,
+      limit: paginationLimit,
```

Comment:
not sure how this is supposed to be used.. how will the caller of fetchStxAddressData know whether they've paginated through all of confirmedTransactions, mempoolTransactions and transferTransactions?

Reply:
good question: the APIs return a "total" value available. so it can be calculated. but for i have seen we aren't not using pagination on mobile and extension, right? 
these changes just propagates the URL params to the inner methods

Reply:
does it solve the original bug?

I think we need to mark this function for refactor. 

- seems like getTransferTransactions returns a superset of getConfirmedTransactions, so not sure why we need to call both.
- the existing logic for adding transfer transactions to confirmed transactions doesn't account for pagination
- mempoolCount should be its total count including unpaginated pendingTransaction items

Reply:
the original issue on mobile won't be solved by these changes. cause was decided to not show the same tx that extension shows. It was decided to only increase the limit. 

the refactor requires additional discussion and work. it seems out of scope for this ticket, but i'm open to discuss if needed.

Reply:
@fedeerbes ok, does the other PR rely on this change? how about we put a comment in the code that this function has some known weird logic in case we see future bugs around pagination on this function

Reply:
hey @teebszet i added a comment referencing this discussion for the future refactor https://github.com/secretkeylabs/xverse-core/pull/217/files#diff-deb0f3d60782723a978c7d4418ac3cc6d144b9f8682c382666d4ab6d7f335e33R55